contracts/extensions/BasicIssuanceExtension.sol

```diff
@@ -0,0 +1,248 @@
+
+/**
+ * @title BasicIssuanceExtension
+ * @author Set Protocol
+ *
+ * Smart contract global extension which provides DelegatedManager owner and 
+ * methodologist the ability to accrue and split issuance and redemption fees at an mutable percentage.
+ */
+contract BasicIssuanceExtension is BaseGlobalExtension {
+    using Address for address;
+    using PreciseUnitMath for uint256;
+    using SafeMath for uint256;
+
+    /* ============ Events ============ */
+
+    event ExtensionInitialized(
+        address _setToken,
+        address _delegatedManager
+    );
+
+    event ExtensionRemoved(
+        address _setToken,
+        address _delegatedManager
+    );
+
+    event FeesDistributed(
+        address _setToken,
+        address indexed _ownerFeeRecipient,
+        address indexed _methodologist,
+        uint256 _ownerTake,
+        uint256 _methodologistTake
+    );
+
+    /* ============ State Variables ============ */
+
+    // Instance of BasicIssuanceModule
+    IIssuanceModule public immutable issuanceModule;
+
+    // Mapping from Set Token to DelegatedManager 
+    mapping(ISetToken => IDelegatedManager) public setManagers;
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        IManagerCore _managerCore,
+        IIssuanceModule _issuanceModule
+    )
+        public
+        BaseGlobalExtension(_managerCore)
+    {
+        issuanceModule = _issuanceModule;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ANYONE CALLABLE: Distributes fees accrued to the DelegatedManager. Calculates fees for
+     * owner and methodologist, and sends to owner fee recipient and methodologist respectively.
+     */
+    function distributeFees(ISetToken _setToken) public {
+        IDelegatedManager delegatedManager = _manager(_setToken);
+
+        uint256 totalFees = _setToken.balanceOf(address(delegatedManager));
+
+        address methodologist = delegatedManager.methodologist();
+        address ownerFeeRecipient = delegatedManager.ownerFeeRecipient();
+
+        uint256 ownerTake = totalFees.preciseMul(delegatedManager.ownerFeeSplit());
+        uint256 methodologistTake = totalFees.sub(ownerTake);
+
+        if (ownerTake > 0) {
+            delegatedManager.transferTokens(address(_setToken), ownerFeeRecipient, ownerTake);
+        }
+
+        if (methodologistTake > 0) {
+            delegatedManager.transferTokens(address(_setToken), methodologist, methodologistTake);
+        }
+
+        emit FeesDistributed(address(_setToken), ownerFeeRecipient, methodologist, ownerTake, methodologistTake);
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external {
+        ISetToken setToken = _delegatedManager.setToken();
+
+        require(
```

Comment:
Instead of this convoluted logic where we are trying to implicitly figure out if the manager is a valid manager created by one of our factories lets keep a state on the Core contract that tracks all managers deployed by the factory. The we can just check that `msg.sender == manager.owner()`.

Reply:
we're not figuring out if the manager is a valid manager created by one of our factories. 

we are figuring out if we have either 
- a manager of a setToken calling
- a factory calling